chimedb/dataflag/client.py

```diff
@@ -144,11 +195,395 @@ def convert(self, value, param, ctx):
+    """List known flagging opinion types."""
+    for type_ in orm.DataFlagOpinionType.select():
+        click.echo(type_.name)
+
+
+@opinion_type.command("create")
+@click.argument("name")
+@click.option(
+    "--description", help="Description of flagging opinion type.", default=None
+)
+@click.option(
+    "--metadata", type=JSON, help="Extra metadata as as JSON dict.", default=None
+)
+@click.option("--force", "-f", is_flag=True, help="Create without prompting.")
+def create_opinion_type(name, description, metadata, force):
+    """Create a new data flag opinion type with given NAME, and optional description and metadata.
+    """
+    type_ = orm.DataFlagOpinionType()
+
+    type_.name = name
+    type_.description = description
+    type_.metadata = metadata
+
+    if force:
+        type_.save()
+    else:
+        click.echo("Type to create:\n")
+        click.echo(format_type(type_))
+        if click.confirm("Create type?"):
+            type_.save()
+            click.echo("Success.")
+        else:
+            click.echo("Aborted.")
+
+
+@opinion_type.command("show")
+@click.argument("opinion_type", type=OTYPE, metavar="TYPE")
+def show_opinion_type(opinion_type):
+    """Show details of the specified opinion TYPE."""
+    click.echo(format_type(opinion_type))
+
+
+@cli.group("opinion")
+def opinion():
+    """Insert data flagging opinions."""
+    pass
+
+
+@opinion.command("create")
+@click.argument("type_", type=OTYPE, metavar="TYPE")
+@click.argument("start", type=TIME)
+@click.argument("finish", type=TIME)
+@click.argument(
+    "decision",
+    type=click.Choice(orm.DataFlagOpinion.decision.enum_list, case_sensitive=False),
+)
+@click.option("--description", help="Description of flag.", default=None)
+@click.option(
+    "--user",
+    "-u",
+    help="Wiki user adding the flag opinion. If no user is supplied, the name of the local user is used.",
+    default=None,
+)
+@click.option(
+    "--instrument",
+    type=click.Choice(["chime", "pathfinder"]),
+    help="Name of instrument to apply flag to.",
+    default=None,
+)
+@click.option("--freq", type=FREQ, help="List of frequency IDs to flag.", default=None)
+@click.option("--inputs", type=INPUTS, help="List of input IDs to flag.", default=None)
+@click.option(
+    "--metadata", type=JSON, help="Extra metadata as as JSON dict.", default=None
+)
+@click.option("--revision", "-r", type=REVISION, required=True)
+@click.option("--force", "-f", is_flag=True, help="Create without prompting.")
+def create_opinion(
+    type_,
+    start,
+    finish,
+    decision,
+    instrument,
+    description,
+    user,
+    freq,
+    inputs,
+    metadata,
+    revision,
+    force,
+):
+    """Create a new data flagging opinion with given TYPE and START and FINISH times.
+
+    Times can be supplied in any format recognized by the `arrow` library. Using
+    the YYYY-MM-DDTHH:MM:SSZ format is recommended. If the flag has not ended,
+    supply the string 'null' instead.
```

Comment:
```suggestion
    supply the string 'null' or 'none' instead.
```
Both are allowed (to accommodate both SQL- and Python- minded people).